### DISCUSSION TITLE
Additional modes for "Add Arc" tool

### DISCUSSION THREAD

**Discussion by hlorus** [hlorus/CAD_Sketcher#82] [Feature Requests]
It's sometimes a bit cumbersome to always place/pick a center point first. It would be nice to have a tool/mode that lets you begin with the startpoint and set the radius in the end.

The main question is probably how the location of the centerpoint could be guessed while drawing the endpoint.

**Comment by hlorus** [comment]
There could also be a mode/option to tangentially constrain the arc to the connecting entity from the starting point. However if there are multiple entities connected it might not be obvious which one to pick.

**Comment by TheMakerTales** [comment]
Well, this is what FreeCAD has.... Ultimately I do find that it's normally called a 3 point arc so it sort of makes sense.
I don't really know if there is any other way unless you start to take in the idea of bezier curves and handles but those in my opinion are very hard to control through precision.

![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/103122109/165989019-31b98033-c820-4242-a20c-ca631d90502a.gif)

**Reply by Intuos5** [reply]
There could also be a Start - End - Radius option.

**Reply by TheMakerTales** [reply]
start and end?
I don't see how that would be implemented through constraints. If you could illustrate how I would be interested to see the setup

**Reply by Intuos5** [reply]
In Autocad, the arc midpoint is assumed to be in between the start and end points (this is also the case in Fusion 360):

https://user-images.githubusercontent.com/63149655/167859426-8ad1fde2-0ac5-4c22-8e3b-d09276fce996.mp4

For sketching purposes, this would be neat as it is rather quick to use. 

There's also the option to do a tangent arc, so that the arc is drawn tangent to the line/ curve from which the start point is used. 

https://user-images.githubusercontent.com/63149655/167860447-be6e336a-6d37-4298-8c60-6f3c382d6908.mp4
